### PR TITLE
Purge script for cleaning out old services from cachet

### PR DESCRIPTION
**Pull request by thearifismail**
Added script for cleaning out cachet

### CODE REVIEW

**Review by kylape — APPROVED** [review]
lgtm.  The only think I can see is that maybe you should check the status code coming from Cachet, but seeing that this is a first pass at a script, it's not a big deal.

**Comment by thearifismail** [comment]
What does it mean by checking status code from Cachet?  Do you mean after deleting components and groups from Cachet, I should ask Cachet one last time, if it has any components or groups left?

**Comment by kylape** [comment]
No, I mean check the HTTP response status code before reading the data from the response.

File: cachet-tools/purge-cachet.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+HEADERS = { 
+    'X-Cachet-Token': os.environ.get("CACHET_TOKEN")
+}
+
+with requests.Session() as session:
+    session.headers.update(HEADERS)
+
+    response = session.get(URL  + "/groups", verify=False)
+
+    if (response.status_code >= 200 and response.status_code <= 299):
```
Comment (kylape) [inline cachet-tools/purge-cachet.py]: I would invert the if clause, e.g. `if (response.status_code >= 300):` and then raise an exception or something.  That reduces the indention for all this code.  And you'll want to do the same thing for your other requests.

File: cachet-tools/purge-cachet.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        groups   = response.json()['data']
+        print("Number of groups found: " + str(len(groups)))
+
+        for group in groups:
+            components = group['enabled_components']
+            print(group['name'] + " contains " + str(len(components)) + " components")
+            for component in components:
+                print("Deleting component: " + component['name'])
+                cdr = session.delete(URL + "/" + str(component['id']), verify=False, )
```
Comment (kylape) [inline cachet-tools/purge-cachet.py]: check it here

File: cachet-tools/purge-cachet.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                print (cdr)
+                # delete the group
+                print("Deleting group " + group['name'])
+                gdr = session.delete(URL + "/groups/" + str(group['id']), verify=False, )
```
Comment (kylape) [inline cachet-tools/purge-cachet.py]: check it here

File: cachet-tools/purge-cachet.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+                print(gdr)
+
+        # check and delete components not in any groups
+        response = session.get(URL, verify=False)
```
Comment (kylape) [inline cachet-tools/purge-cachet.py]: check it here

File: cachet-tools/purge-cachet.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        components = response.json()['data']
+        print("Number of components not in any group: " + str(len(components)))
+
+        for component in components:
+            print("Deleting component: " + component['name'])
+            cdr = session.delete(URL + "/" + str(component['id']), verify=False, )
```
Comment (kylape) [inline cachet-tools/purge-cachet.py:25]: check it here

**Review by kylape — CHANGES_REQUESTED** [review]
If you're gonna add the check, please add it to all the requests.

**Comment by thearifismail** [comment]
> If you're gonna add the check, please add it to all the requests.

If I don't add the check what other solution you have in mind?

**Review by kylape — APPROVED** [review]

**Pull request merged**